release_notes/versioning-policy.adoc

```diff
@@ -0,0 +1,48 @@
+[id='ocp-4-0-versioning-policy']
+= {product-title} Versioning Policy
+include::modules/common-attributes.adoc[]
+:context: release-notes
+
+toc::[]
+
+{nbsp} +
+{product-title} provides strict backwards compatibility guarantees for all
+supported APIs, excluding alpha APIs (which may be changed without notice) and
+beta APIs (which may occasionally be changed in a non-backwards compatible
+manner).
+
+The {product-title} version must match between master and node hosts, excluding
+temporary mismatches during cluster upgrades. For example, in a 3.11 cluster, all
+masters must be 3.11 and all nodes must be 3.11. However, {product-title} will
+continue to support older `oc` clients against newer servers. For example, a 3.5
+`oc` will work against 3.4, 3.5, and 3.6 servers.
+
+Changes of APIs for non-security related reasons will involve, at minimum, two
+minor releases (3.4 to 3.5 to 3.6, for example) to allow older `oc` to update.
+Using new capabilities may require newer `oc`. A 3.2 server may have additional
+capabilities that a 3.1 `oc` cannot use and a 3.2 `oc` may have additional
+capabilities that are not supported by a 3.1 server.
+
+.Compatibility Matrix
+
+[cols="1,1,1"]
+|===
+
+|
+|*X.Y* (`oc` Client)
+|*X.Y+N* footnoteref:[versionpolicyn,Where *N* is a number greater than 1.] (`oc` Client)
+
+|*X.Y* (Server)
+|image:redcircle-1.png[]
+|image:redcircle-3.png[]
+
+|*X.Y+N* footnoteref:[versionpolicyn] (Server)
+|image:redcircle-2.png[]
+|image:redcircle-1.png[]
+
+|===
+image:redcircle-1.png[] Fully compatible.
+
+image:redcircle-2.png[] `oc` client may not be able to access server features.
+
+image:redcircle-3.png[] `oc` client may provide options and features that may not be compatible with the accessed server.
```

Comment:
Above is copied from 3.x document. Its content is right. As a 4.x note, it doesn't mention any 4.x word. It is better to mention that. E.g. 4.1 and 3.11 is compatible (4.0 is skipped because it is not a release version)

Reply:
@xingxingxia See above,
{product-title} 3.11 is compatible with {product-title} {product-version}. Red
Hat did not publicly release {product-title} 4.0 and, instead, is releasing
{product-title} {product-version} directly after version 3.11.

Currently, it renders to,
http://file.rdu.redhat.com/~ahardin/05072019/4-0-release-notes/release_notes/versioning-policy.html
OpenShift Container Platform 3.11 is compatible with OpenShift Container Platform Branch Build. Red Hat did not publicly release OpenShift Container Platform 4.0 and, instead, is releasing OpenShift Container Platform Branch Build directly after version 3.11.

"Branch build" is used for "{product-version}", but I think it will use "4.1" when release.
Then you will see,
OpenShift Container Platform 3.11 is compatible with OpenShift Container Platform 4.1. Red Hat did not publicly release OpenShift Container Platform 4.0 and, instead, is releasing OpenShift Container Platform 4.1 directly after version 3.11.

Reply:
@liangxia Correct. Thank you!